src/Monitoring/Unit.cpp

```diff
@@ -0,0 +1,114 @@
+#include "Unit.hpp"
+
+#include <ios>
+#include <string>
+#include <vector>
+#include <cmath>
+#include <sstream>
+
+namespace {
+const static std::vector<std::string> PositivePrefixes = {
+    "k", "M", "G", "T", "P", "E", "Z", "Y", "R", "Q"};
+const static std::vector<std::string> PositiveBytePrefixes = {
+    "Ki", "Mi", "Gi", "Ti", "Pi", "Ei", "Zi", "Yi"};
+const static std::vector<std::string> NegativePrefixes = {
+    "m", "µ", "n", "p", "f", "a", "z", "y", "r", "q"};
+} // namespace
+
+namespace seissol {
+SIUnit::SIUnit(const std::string& unit, bool binary) : unit(unit), binary(binary) {}
+
+std::string SIUnit::formatTime(double value, bool exact, int digits) const {
+  const double byDay = std::floor(value / (60 * 60 * 24));
+  const double hours = value - byDay * (60 * 60 * 24);
+  const double byHour = std::floor(hours / (60 * 60));
+  const double minutes = hours - byHour * (60 * 60);
+  const double byMinute = std::floor(minutes / 60);
+  const double seconds = minutes - byMinute * (60);
+
+  std::ostringstream stream;
+  stream.precision(0);
+  stream << std::fixed;
+  bool done = false;
+  if (byDay > 0 && (!done || exact)) {
+    stream << byDay << " d";
+    done = true;
+  }
+  if (byHour > 0 && (!done || exact)) {
+    if (done) {
+      stream << " ";
+    }
+    stream << byHour << " h";
+    done = true;
+  }
+  if (byMinute > 0 && (!done || exact)) {
+    if (done) {
+      stream << " ";
+    }
+    stream << byMinute << " min";
+    done = true;
+  }
+  if (seconds != 0 || !done || exact) {
+    if (done) {
+      stream << " ";
+    }
+    stream << formatPrefix(seconds, digits);
+    done = true;
+  }
+  return stream.str();
+}
+
+std::string SIUnit::formatPrefix(double value, int digits) const {
+  double adjValue = std::abs(value);
```

Comment:
what means adj?

Reply:
I think I thought of "adjusted value" or so when writing this; probably mantissa or so is better?

Reply:
ok, maybe mantissa would be better.